lib/src/configuration.rs

```diff
@@ -0,0 +1,59 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+}
+
+// Check whether defining BasicAuth works like this or not, else revert to the basic definition commented out
+#[derive(Debug, Clone)]
+pub struct BasicAuth {
+    pub username: String,
+    pub password: Option<String>,
+}
+// pub type BasicAuth = (String, Option<String>);
+
+#[derive(Debug, Clone)]
+pub struct ApiKey {
+    pub prefix: Option<String>,
+    pub key: String,
+}
+
+impl Configuration {
+    pub fn new(
+        base_path: String,
```

Comment:
Could perhaps use https://crates.io/crates/url?

Reply:
I like this idea! Changed